### PR TITLE
feat: opt-in for access logs

### PR DESCRIPTION
**Pull request by alovh**

### CODE REVIEW

**Review by AlexanderBrevig — CHANGES_REQUESTED** [review]

File: modules/bucket/main.tf [hunk 1]
```diff
@@ -2,6 +2,7 @@ locals {
   generation       = format("%03d", var.generation)
   bucket_shortname = var.name_override != null ? var.name_override : var.init.app.id
   bucket_name      = "ent-gcs-${local.bucket_shortname}-${var.init.environment}-${local.generation}"
+  log_bucket_name  = var.enable_access_logs ? "ent-gcs-${local.bucket_shortname}-${var.init.environment}-${local.generation}-access-logs" : ""
```
[the remark this answers is not in the document]
Comment (AlexanderBrevig) [reply modules/bucket/main.tf:5]: https://enturas.atlassian.net/wiki/spaces/ESP/pages/3182264469/Naming+conventions+on+GCP

File: examples/minimal_test/main.tf [hunk 2]
```diff
@@ -19,8 +19,9 @@ module "cloud-storage" {
...
   # source     = "github.com/entur/terraform-google-cloud-storage//modules/cloud-storage?ref=vVERSION"
-  source = "../../modules/bucket"
-  init   = module.init
-  generation = var.generation != null ? var.generation : random_integer.random_revision_generation.result
-  name_override = var.name_override != null ? var.name_override : var.init.app.id
+  source             = "../../modules/bucket"
+  init               = module.init
+  generation         = var.generation != null ? var.generation : random_integer.random_revision_generation.result
+  name_override      = var.name_override != null ? var.name_override : var.init.app.id
+  enable_access_logs = var.enable_access_logs != null ? var.enable_access_logs : true
```
[the remark this answers is not in the document]
Comment (alovh) [reply examples/minimal_test/main.tf:26]: haha, her har jeg tenkt litt for hardt ja XD

File: modules/bucket/main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (alovh) [reply modules/bucket/main.tf:5]: made it a vote :)

File: modules/bucket/variables.tf [hunk 3]
```diff
@@ -1,3 +1,8 @@
+variable "enable_access_logs" {
+  description = "Set to true to enable access logs, log bucket will have bucket shortname appended with 'access-logs'. Bucket 'ent-gcs-foo-dev-001' will have access logs in bucket 'ent-gcs-foo-access-logs-dev-001'"
```
Comment (AlexanderBrevig) [inline modules/bucket/variables.tf:2]: doc does not match the test case `"ent-gcs-%s-axlogs-dev-%s"`

**Review by AlexanderBrevig — CHANGES_REQUESTED** [review]
Did we figure out if there exists a default log bucket? Should we use that if that's the case?

**Comment by alovh** [comment]
> Did we figure out if there exists a default log bucket? Should we use that if that's the case?

funny thing, there is, but we cannot actively send logs to the log buckets, they must be sent to a storage bucket. And a default log storage bucket does not exist. added a comment in the issue btw

File: modules/bucket/variables.tf (hunk 3 above)
Comment (AlexanderBrevig) [reply modules/bucket/variables.tf:2]: maybe a simple `logs` infix is better? Shorter than `access-logs` and easier to understand than `axlogs`?

Comment (AlexanderBrevig) [reply modules/bucket/variables.tf:2]: There's a mismatch between what's documented and what actually happens.
What do you think about the suggestion of using only a simple `logs` infix?

**Pull request closed without merging**